rasa/nlu/classifiers/keyword_intent_classifier.py

```diff
@@ -1,32 +1,119 @@
+    from rasa.nlu.model import Metadata
+    from rasa.nlu.training_data import Message
+
 
 class KeywordIntentClassifier(Component):
+    """Intent classifier using simple keyword matching.
+
+
+    The classifier takes a list of keywords and associated intents as an input.
+    A input sentence is checked for the keywords and the intent is returned.
+
+    """
 
     provides = ["intent"]
 
-    his = ["hello", "hi", "hey"]
+    defaults = {"case_sensitive": True}
 
-    byes = ["bye", "goodbye"]
+    def __init__(
+        self,
+        component_config: Optional[Dict[Text, Any]] = None,
+        intent_keyword_map: Optional[Dict] = None,
+    ):
 
-    def process(self, message: Message, **kwargs: Any) -> None:
+        super(KeywordIntentClassifier, self).__init__(component_config)
 
-        intent = {"name": self.parse(message.text), "confidence": 1.0}
-        message.set("intent", intent, add_to_output=True)
+        if intent_keyword_map is None:
+            self.intent_keyword_map = {}
+        else:
+            self.intent_keyword_map = intent_keyword_map
 
-    def parse(self, text: Text) -> Optional[Text]:
+    def train(
+        self,
+        training_data: "TrainingData",
+        cfg: Optional["RasaNLUModelConfig"] = None,
+        **kwargs: Any
+    ) -> None:
 
-        _text = text.lower()
+        for intent in training_data.intents:
+            self.intent_keyword_map[intent] = [
+                ex.text
+                for ex in training_data.training_examples
+                if ex.get("intent") == intent
+            ]
 
-        def is_present(x):
-            return x in _text
+    def process(self, message: Message, **kwargs: Any) -> None:
+        intent_name = self._map_keyword_to_intent(message.text)
+        if intent_name is not None:
+            intent = {"name": intent_name, "confidence": 1.0}
+            message.set("intent", intent, add_to_output=True)
 
-        if any(map(is_present, self.his)):
-            return "greet"
-        elif any(map(is_present, self.byes)):
-            return "goodbye"
-        else:
+    def _map_keyword_to_intent(self, text: Text) -> Optional[Text]:
+        found_intents = []
+        for intent, examples in self.intent_keyword_map.items():
+            for example in examples:
+                if self.component_config["case_sensitive"]:
+                    if example in text:
+                        found_intents.append(intent)
+                else:
+                    if example.lower() in text.lower():
+                        found_intents.append(intent)
+        if len(found_intents) == 0:
             return None
+        elif len(found_intents) == 1:
+            return found_intents[0]
+        else:
+            logger.debug(
+                "KeywordClassifier found keywords for intents '{}',"
+                "will classify message as having intent '{}'."
+                "".format(found_intents, found_intents[0])
```

Comment:
I think it is better to raise this warning during training, saying smth like there are ambiguous intents ...

Reply:
Good idea, I added a warning during training, but this is a different case. This debug message is logged in case a user's message contains several keywords. In that case the intent classification is not reliable (i think it depends on the order of dictionary keys, which is not documented to be fixed in python<3.7)

Reply:
I see let's not predict anything if there are several keywords

Reply:
That means `None` with confidence `0.0` to trigger fallback policy right?

Reply:
yes, I think it's make more sense rather than one of the intents